src/config.h

```diff
@@ -73,6 +73,17 @@
  */
 #ifndef USE_CPU_ONLY
 #define USE_OPENCL
+
+/*
+ * USE_HALF: Include the half-precision OpenCL implementation when building.
+ * Unlike previous versions, this does not enable half-precision by default,
+ * it just compiles half-precision support.  You have to use the command line
+ * argument --use-half explicitly to enable half-precision.
+ * half-precision OpenCL gains performance on some GPUs while losing some
```

Comment:
Obvious question: should we check it during tuning and automatically enable if beneficial?

Reply:
#ifdef USE_HALF
const auto TUNER_KERNEL = std::string("XgemmBatchedHalf");
#endif

This stuff no longer works correctly. (Github didn't let me comment on the actual line)

Reply:
Thinking about autodetecting between fp16 and fp32, I think the best I can possibly try is:
- Do a netbench on both fp16 and fp32, and use fp16 only if fp16 is at least 5% faster.  5% is a bit arbitrary, but last time we tried engine strength was roughly equivalent with same playouts, so 5% gives a bit of margin.
- If fp16 selfcheck fails revert to fp32 - the revert process may be tricky due to possibility of violating time constraint if it were to happen in the middle of a game, though.  Hopefully this doesn't happen too frequently that it loses by time.
- autodetect is default but also allow forcing fp16 or fp32
Any better ideas?

Reply:
>If fp16 selfcheck fails revert to fp32 

Not sure we need to handle this case with the relaxed settings.